### ISSUE TITLE
Search issue with large file of single line

### ISSUE THREAD

**Issue by dakaraphi** [Microsoft/vscode#24264]
This may be an edge case, but there may be something here to be improved.

I encountered some searches taking a very large amount of time compared to others which were nearly instant.  Tracked this down to a 3meg .map file that was appearing in results.

Looks like the entire file might possibly get sent back by ripgrep since it is one line.  Not sure if there is an option to trim the result to some reasonable length so that VS Code could better handle.

Work around of course is to simply exclude such files.

- VSCode Version: 1.11

**Comment by BurntSushi** [comment]
ripgrep has a `-M/--max-columns` switch that may be useful for this.

**Comment by roblourens** [comment]
Yeah, I've seen this. From what I remember, the slowdown is mainly due to sending the entire "line preview" in a single IPC message between the search process and the vscode frontend. I didn't do anything about it because it's no slower than it was before, and people don't usually search files that like.

It would be easy to trim the preview to the 50 chars or so that can actually be displayed, but multiple results on one line use the same preview, so there is some optimal piece or set of pieces of the line to send over.

Also, Node's IPC mechanism is actually much slower than just using a normal socket. I experimented with using that for communication with the search process and found it to give a nice improvement in this case, so maybe I can switch to that in the future. We did the same thing for the extension host process recently.

**Comment by dakaraphi** [comment]
I think I just hadn't noticed before.  Now that `ripgrep` is so crazy fast.  A slow search just stands out more in comparison :-)
